### ISSUE TITLE
SRe2L++ reproducing  for resnet-50 on tiny-imagenet

### ISSUE THREAD

**Issue by sillytheif** [Jiacheng8/CV-DD#6]
Hello, I'd like to ask a quick question about reproducing your results.

We are primarily using SRe2L++ as our baseline. We repeated the full 'recover-relabel-validate' process three times and averaged the results reported at the last epoch of the validation process.

On ResNet-18, the average error we observed is generally within 1 to 2 percentage points of your reported results, which we consider reasonable given the randomness.

However, on ResNet-50, the error is significantly larger. In the ipc=50 case, we only achieved an average accuracy of 42%, which is a very large gap from the 53.7% you reported. Furthermore, upon observing the synthesized images, we noticed they were irregular and also quite different from the images generated with ResNet-18. Our result is closer to the 42% accuracy reported by the original SRe2L paper. 

I would like to ask for your insights on the possible reasons for this discrepancy

**Comment by sillytheif** [comment]
One of our primary suspicions is the hyperparameter settings. Specifically, the weight for the BN (Batch Normalization) loss, r_bn. Intuitively, the BN loss value for ResNet-50 seems excessively large, on the order of tens of thousands, whereas for ResNet-18 it's only in the hundreds. This suggests that the r_bn weight for ResNet-50 probably shouldn't be the same as ResNet-18's, which is 0.01. We would like to hear your opinion on this

**Comment by Jiacheng8** [comment]
Hi, in the post-evaluation phase with ResNet-50 as the student model, we use ResNet-18 as the recover model to generate the distilled data, and the teacher model to produce the soft labels. From your description, it seems that you used ResNet-50 for both the recover model and the teacher model, which I believe is the source of the discrepancy.

**Comment by sillytheif** [comment]
> Hi, in the post-evaluation phase with ResNet-50 as the student model, we use ResNet-18 as the recover model to generate the distilled data, and the teacher model to produce the soft labels. From your description, it seems that you used ResNet-50 for both the recover model and the teacher model, which I believe is the source of the discrepancy.

Thanks for your reply. I'd like to confirm the terminology to ensure we're on the same page. Is my understanding correct that for the Tiny ImageNet dataset for ResNet-50:

The recover process uses a ResNet-50 model.

The relabel process uses a ResNet-18 model to generate soft labels.

The validate process uses a ResNet-18 model to evaluate the performance of the generated soft labels on the corresponding data

**Comment by Jiacheng8** [comment]
Hi, I would like to clarify that all datasets distilled by Sre2L++ are generated by using ResNet-18, which utilizes ResNet-18 to recover and relabel. Then if you want to evaluate on ResNet-50, you evaluate the generated dataset using ResNet-50.

**Comment by sillytheif** [comment]
Thank you for your reply, I think I get it now. Thanks again for your effort and enthusiasm.
